core/engine_test.go

```diff
@@ -957,11 +957,11 @@ func TestMinIterationDuration(t *testing.T) {
 		require.False(t, engine.IsTainted())
 	}
 
-	// Only 2 full iterations are expected to be completed due to the 1 second minIterationDuration
-	assert.Equal(t, 2.0, getMetricSum(collector, metrics.Iterations.Name))
+	// 16 full iterations are expected to be completed
+	assert.Equal(t, 16.0, getMetricSum(collector, metrics.Iterations.Name))
```

Comment:
Hmm this seems a bit strange, given that iterations are currently interruptible - each VU should only be able to make 3 full 150ms iterations in the allotted 500ms, so shouldn't this be 12?

Reply:
3 iterations per VU will be able to both run the `default` function and get through the sleep introduced by the `minIterationDuration`, but 1 will get through the `default` function will emit the metric and will get stuck in the `minIterationDuration`. 

IMO this is the better behaviour - the iteration actually finished/ran through `default`. It also is probably is never going to be a problem in real usage as we don't actually change whether we actually run the `default` function one more time with this PR. It just changes whether `metrics.Iterations` in particular is emitted before or after the sleep.
As I mentioned in a previous comment - I find this behaviour to be the more accurate one - we also actually emit what the Iteration duration is and not the value of `minIterationDuration`.

 My only problem is ... the name of `minIterationDuration` confusing the whole thing, but after the new executors I would think this will be even less used by anyone practically ;)

Reply:
Ah, right, forgot :blush:  :man_facepalming: The test is how it should be, and you are right, the k6 behavior now makes more sense than before! And yeah, even if the the name `minIterationDuration` isn't ideal, the current behavior is. 

But that means that now we're probably lacking a test that incomplete iterations don't emit the `iterations` metric - say, almost the exact same test like this one, but using `sleep()` instead of `minIterationDuration`, should emit only 12 iterations, which I think is worth confirming with another test.

Reply:
OK, I'll add the missing test, which will help me understand this a bit better... :)

Reply:
@na-- Let me know if 41af3670 matches what you had in mind. The JS is a bit funky :smile:, but I think it cuts down on repetition nicely. I renamed the test to something more generic, and added some comments which hopefully make sense.

Reply:
No, that is the expected and wanted behavior :smile: see how the expected values in the test before your PR were different for `iterations` and `testcounter`